clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -519,6 +524,7 @@ private void fetchOffsetsWithRetries(final OffsetFetchRequestState fetchRequest,
                 log.warn("A duplicated, inflight, request was identified, but unable to find it in the " +
                     "outbound buffer:" + fetchRequest);
             }
+            offsetFetchResultCache.maybeCache(fetchRequest, res);
```

Comment:
this here means that we're caching results instead of inflight request, which I'm worried could be risky. Let's take this sequence:

- call to fetch offsets for tp0 (fetch1) + times out without response
- fetch1 completes after it is expired -> we save the results in cache
- commit offsets for tp0 -> completes successfully
- call to fetch offsets for tp0 again (fetch2) => won't this reuse the cached result received for fetch1 which is not correct?

Reply:
I've added a unit test for this case. Without the commit cache clearing it fails, but with it, it passes.